src/adagrams.js

```diff
@@ -1,15 +1,100 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+  let alphabetPool = 'aaaaaaaaaabbccddddeeeeeeeeeeeeffggghhiiiiiiiiijkllllmmnnnnnnooooooooppqrrrrrrssssttttttuuuuvvwwxyyz'
+  alphabetPool = alphabetPool.toUpperCase()
+  let i = 0;
+  let playerHand = []
+  
+  do {
+    i++;
+    let singleLetter = alphabetPool.charAt(Math.random() * alphabetPool.length);
+    playerHand.push(singleLetter);
+    alphabetPool = alphabetPool.replace(singleLetter, '');
+  } while (i < 10);
+
+  return playerHand;
+}
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let playerHand = {}
+  for (let i of lettersInHand) {
+    if (i in playerHand) {
+      playerHand[i] += 1
+    } else {
+      playerHand[i] = 1
+    }
+  } 
+
+  for (let j of input) {
+    if (!(j in playerHand)) {
+      return false;
+    } else {
+      playerHand[j] -= 1
+    }
+  }
+
+  return Object.values(playerHand).every(
+    value => value >= 0
+  )
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  const scoreChart = {
+    1: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+    2: ['D', 'G'],
+    3: ['B', 'C', 'M', 'P'],
+    4: ['F', 'H', 'V', 'W', 'Y'],
+    5: ['K'],
+    8: ['J', 'X'],
+    10: ['Q', 'Z']
+  };
+  
+  let wordScore = 0;
+  for (let i = 0; i < word.length; i++) {
+    for (const [key, value] of Object.entries(scoreChart)) {
+      for (const v of value) {
+        if (word[i].toUpperCase() === v) {
+          wordScore += Math.floor(key)
+        }
+      }
+    }
+  };
+  if (word.length >= 7) {
+    wordScore += Math.floor(8) 
+  };
```

Comment:
Nice work with this function! I noticed you didn't add a check for if the length of the word is 7, 8, 9, or 10. You only checked for if the word was simply greater than 7, but you need to specify that the word length is also less than 11 to get a score or 8. It looks like this is a mistake on our part when it comes to our tests.